### PR TITLE
Hotfix 214

### LINKED ISSUE #214: bug in flag_complex
Without the `ps` argument, `xgi.flag_complex()` fills all 4 triangles of the complete graph, as expected:

```python
import networkx as nx
import xgi

G = nx.complete_graph(n=4)
S = xgi.flag_complex(G)
xgi.num_edges_order(S, 2)
# -> 4
```
Including it as `ps=[1]` should be equivalent  
```python
S = xgi.flag_complex(G, ps=[1])
xgi.num_edges_order(S, 2)
# -> 0 # huh?
```
We are only computing maximal cliques and feeding them to `add_simplices_from` which might not be adding subfaces as expected?

### PR DESCRIPTION
**Pull request by maximelucas**
- fixed #214 : all simplices are added correctly with max_order now
- added a `powerset` function to compute all subsets of a set
- added a `subfaces` function. 
- added tests and examples for all the above, and included them in docs

The `subfaces` extends the current private `_subfaces` present in `SimplicialComplex.py` with a few options, and is available publicly. I think it should ultimately replace `_subfaces`.

### CODE REVIEW

File: xgi/utils/utilities.py [hunk 1]
```diff
@@ -35,3 +36,42 @@ def dual_dict(edge_dict):
...
+
+    if include_empty:
+        start = 0
+    else:
+        start = 1
+
+    if include_full:
+        end = 1
+    else:
+        end = 0
```
Comment (leotrs) [inline xgi/utils/utilities.py:75]: Nitpick:
```suggestion
    start = 0 if include_empty else 1
    end = 1 if include_fill else 0
```

File: xgi/utils/utilities.py [hunk 2]
```diff
@@ -35,3 +36,42 @@ def dual_dict(edge_dict):
...
+
+    s = list(iterable)
```
Comment (leotrs) [inline xgi/utils/utilities.py:77]: Question: doesn't `combinations` accept an iterable? Why convert to list?

**Review by leotrs — APPROVED** [review]
Approved, feel free to ignore my nitpick.

File: xgi/utils/utilities.py (hunk 2 above)
Comment (maximelucas) [reply xgi/utils/utilities.py:77]: Not sure, took it from itertools recipes (https://docs.python.org/3/library/itertools.html#itertools-recipes).
I'd keep it for safety, but shouldn't change much as the input should be a single edge.

File: xgi/utils/utilities.py (hunk 1 above)
Comment (maximelucas) [reply xgi/utils/utilities.py]: Done

**Pull request merged**